docs/source/new-project.rst

```diff
@@ -56,26 +93,84 @@ A lightningModule defines
 
         def configure_optimizers(self):
             return torch.optim.Adam(self.parameters(), lr=0.0005)
+            
+            
+You can use your :class:`~pytorch_lightning.core.LightningModule` just like a PyTorch model. Read more `here<https://pytorch-lightning.readthedocs.io/en/stable/lightning-module.html>`_.
 
 ----------
 
-Step 2: Fit with a Trainer
---------------------------
-The trainer calls each loop at the correct time as needed. It also ensures it all works
-well across any accelerator.
+Step 3: Define you data loaders
+-------------------------------
+
+:class:`~pytorch_lightning.core.DataModule`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+A :class:`~pytorch_lightning.core.DataModule` is simply a collection of all 3 data splits but
+also captures:
+
+- download instructions.
+- processing.
+- splitting.
+- etc...
+
+.. code-block:: python
+
+    class MyDataModule(pl.DataModule):
+
+        def __init__(self):
+            ...
+
+        def train_dataloader(self):
+            # your train transforms
+            return DataLoader(YOUR_DATASET)
+
+        def val_dataloader(self):
+            # your val transforms
+            return DataLoader(YOUR_DATASET)
+
+        def test_dataloader(self):
+            # your test transforms
+            return DataLoader(YOUR_DATASET)
+
+
+DataModules are specifically useful for building models based on data. Read more `here<https://pytorch-lightning.readthedocs.io/en/stable/datamodules.html>`_.
+
+PyTorch DataLoader
+^^^^^^^^^^^^^^^^^^
+If you don't want to craete a DataModule you can use plain PyTorch :class:`~torch.utils.data.DataLoader`, and add them to your :class:`~pytorch_lightning.core.lightningModule`:
 
 .. code-block:: python
 
-    # dataloader
-    dataset = MNIST(os.getcwd(), download=True, transform=transforms.ToTensor())
-    train_loader = DataLoader(dataset)
+    class LitModel(pl.LightningModule):
+
+        def train_dataloader(self):
+            # your train transforms
+            data_set = MNIST(os.getcwd(), download=True, transform=transforms.ToTensor())
+            return DataLoader(data_set)
+            
+or just use them directly in your trainer.
+
+----------
+
+Step 4: Fit with a Trainer
+--------------------------
+You init your :class:`~pytorch_lightning.core.LightningModule`, your :class:`~pytorch_lightning.core.DataModule` and then the :class:`~pytorch_lightning.core.Trainer`. 
+The Lightning :class:`~pytorch_lightning.core.Trainer` is where the magic happens! It abstracts the most of the training engineering code such as:
+
+* The batch iteration
+* The epoch iteration
+* Calling of optimizer.step()
+
+
+.. code-block:: python
 
     # init model
     model = LitModel()
+    # init data
+    data_module = MyDataModule()
 
-    # most basic trainer, uses good defaults
+    # most basic trainer, uses good defaults and data module
     trainer = pl.Trainer()
-    trainer.fit(model, train_loader)
+    trainer.fit(model, data_module)
```

Comment:
have you tested this code to work? isn't the data_module an extra kwarg coming after the usual data loader?

Reply:
https://pytorch-lightning.readthedocs.io/en/latest/datamodules.html#using-a-datamodule